### DISCUSSION TITLE
Tuya garage door opener (GDC311ZBQ1) not reporting contact  sensor

### DISCUSSION THREAD

**Discussion by LarrxX** [Koenkk/zigbee2mqtt#12716] [Devices]
I am using zigbee2mqtt with Home Assistant, on Raspberry Pi 4 and a Conbee 2 stick.
Everything seems to be OK but I am unable to get the status of the open/closed door wireless sensor of the Tuya garage door opener (GDC311ZBQ1). 

I can correctly send commands to trigger the switch, and the sensor LED blinks when the contact is broken (so I assume it is working).
![Screenshot_2022-06-05-23-17-16-58_c3a231c25ed346e59462e84656a70e50](https://user-images.githubusercontent.com/16936069/172071007-d95967ae-baff-4a4e-ba0e-a1a5b433529b.jpg)
 

Thanks

**Comment by LarrxX** [comment]
I am not afraid to go down into the nitty gritty stuff, but I am new to HA and Zigbee2mqtt. I just need a nudge in the correct direction.

**Comment by LarrxX** [comment]
OK, so it suddenly started working by itself. Weird....

**Reply by grest** [reply]
in my case it's reporting always closed. Any idea how to fix it?

**Reply by LarrxX** [reply]
No, sorry. It just started working, and I didn't do anything

**Reply by grest** [reply]
after re-pairing it became unavailable; isn't visible in the state json. I'll report the bug probably. 

How long did it take to start working?

**Reply by grest** [reply]
@LarrxX and what kind of device do you have? Is it orange one, made by loratap?

**Reply by LarrxX** [reply]
I just realized this is a post for Zigbee2mqtt. I actually stopped using Zigbee2mqtt and am using ZHA directly.

**Reply by LarrxX** [reply]
But yes, it is the device you are describing, and it did start working on ZIgbee2mqtt all by itself, back then.
Sorry I can't help more.

**Comment by grest** [comment]
@LarrxX, I found a solution in my case. I had to to re-pair the wireless sensor with the main device. After pairing the main device with zigbee2mqtt & HA, I repeated the pairing procedure, this time keeping the reset button longer, even after the light started blinking red already. At some point (2-3 seconds later), it blinked blue twice. Then I started to play with the wireless sensor, and suddenly the whole device disappeared and appeared again in zigbee2mqtt, this time with wireless sensor working.
